packages/appium/lib/appium.js

```diff
@@ -167,7 +167,13 @@ class AppiumDriver extends BaseDriver {
       let runningDriversData, otherPendingDriversData;
       const parsedInnerDriverArgs = parseExtensionArgs(this.args.driverArgs, driverName);
 
-      const d = new InnerDriver(this.args, true, parsedInnerDriverArgs);
+      if (_.has(InnerDriver, 'argsConstraints')) {
+        const driverArgsConstraints = InnerDriver.argsConstraints;
+        // parse/verify CLI args and merge them to this.args if OK or throw an exception
+        parseDriverPluginArgs(this.args, parsedInnerDriverArgs, driverArgsConstraints);
```

Comment:
I would prefer this method to return parsed args rather than mutate them

Reply:
it might also be useful to add a debug log here about which args/values were parsed